core/base/src/TROOT.cxx

```diff
@@ -1558,16 +1558,22 @@ TStyle *TROOT::GetStyle(const char *name) const
 
 TObject *TROOT::GetFunction(const char *name) const
 {
-   if (name == nullptr || name[0] == 0) {
+   if (!name || !*name)
       return nullptr;
-   }
 
    {
       R__LOCKGUARD(gROOTMutex);
       TObject *f1 = fFunctions->FindObject(name);
       if (f1) return f1;
    }
 
+   static bool _init_standard = false;
+
+   if (_init_standard)
+      return nullptr;
+
+   _init_standard = true;
+
```

Comment:
I propose the following full implementation (`fFunctions` is now a thread safe container)
```C++
   if (!name || !*name)
      return nullptr;

   TObject *f1 = fFunctions->FindObject(name);
   if (f1)
      return f1;

   // This will induces the loading of `libHist` by placing it here
   // we delay the loading until there is a search for a not (yet)
   // known function.
   auto init_TF1_standard = [&f1]() {
      gROOT->ProcessLine("TF1::InitStandardFunctions();");
      f1 = fFunctions->FindObject(name);
      return true;
  };
  static bool init = init_TF1_standard();
  return f1;
```